WordPress/Sniffs/PHP/DiscouragedFunctionsSniff.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Discourages the use of various native PHP functions and suggests alternatives.
+ *
+ * @package    WPCS\WordPressCodingStandards
+ *
+ * @since      0.1.0
+ * @since      0.10.0 The checks for the POSIX functions have been replaced by the stand-alone
+ *                    sniff WordPress_Sniffs_PHP_POSIXFunctionsSniff.
+ * @deprecated 0.11.0 The checks for the PHP development functions have been replaced by the
+ *                    stand-alone sniff WordPress_Sniffs_PHP_DevelopmentFunctionsSniff.
+ *                    The checks for the WP deprecated functions have been replaced by the
+ *                    stand-alone sniff WordPress_Sniffs_WP_DeprecatedFunctionsSniff.
+ *                    The checks for the PHP functions which have a WP alternative has been replaced
+ *                    by the stand-alone sniff WordPress_Sniffs_WP_AlternativeFunctionsSniff.
+ *                    The checks for the WP discouraged functions have been replaced by the
+ *                    stand-alone sniff WordPress_Sniffs_WP_DiscouragedFunctionsSniff.
+ *                    by the stand-alone sniff WordPress_Sniffs_WP_AlternativeFunctionsSniff.
+ *                    The checks for the PHP discouraged functions have been replaced by the
+ *                    stand-alone sniff WordPress_Sniffs_WP_DiscouragedPHPFunctionsSniff.
+ *                    The check for the `register_globals` has been removed as there is no such
+ *                    function. To check for `register_globals` ini directive use
+ *                    PHPCompatibility_Sniffs_PHP_DeprecatedIniDirectivesSniff from wimg/PHPCompatibility.
+ */
+class WordPress_Sniffs_PHP_DiscouragedFunctionsSniff extends WordPress_AbstractFunctionRestrictionsSniff {
```

Comment:
If you remove the `extends WordPress_AbstractFunctionRestrictionsSniff`, you can remove the everything else in the class too, making it even clearer that it's deprecated.